### PR TITLE
Feat/dq improve current icons in the stepper

### PR DESCRIPTION
**Pull request by mariaozamiz**
### :pushpin: References

-   **Issue:** Closes [DQ: Improve current icons in the Stepper](https://app.clickup.com/t/86947x6dc)
- CU-86947x6dc

### :memo: Implementation

The color scheme is supposed to work like this:
- The step is either pending of analysis or active --> grey background
- The step has unresolved issues --> red background
- The step has no issues --> green background

As client is about to check the app, I left some classes there just in case there is need of changing the color scheme. 

### :video_camera: Screenshots/Screen capture
Captura de pantalla 2024-04-09 a las 14 14 12

### CODE REVIEW

**Comment by ifoche** [comment]
Task linked: [CU-86947x6dc DQ: Improve current icons in the Stepper](https://app.clickup.com/t/86947x6dc)

**Review by Ramon-Jimenez — CHANGES_REQUESTED** [review]
Only one small change, when step is active but it has been run, keep the bold but also the red/green colors.

So by using text in bold we represent the active step and by using grey/red/green we represent steps not run/issues found/issues not found.

If this doesn't make sense, let's quickly catch up on Slack. Thanks @mariaozamiz !

File: src/webapp/pages/analysis/AnalysisPage.tsx [hunk 1]
```diff
@@ -64,11 +91,19 @@ function buildStepsFromSections(
 
             const index = analysis.sections.findIndex(s => s.id === section.id) + 1;
             const isActive = currentSection === section.name;
-            const isCompleted = !QualityAnalysisSection.isPending(section);
+            const isPending = section.status === "pending";
```
Comment (eperedo) [inline src/webapp/pages/analysis/AnalysisPage.tsx:94]: we can keep the method but without the NOT (!) operator.

```
const isPending = QualityAnalysisSection.isPending(section);
```

File: src/webapp/pages/analysis/AnalysisPage.tsx [hunk 2]
```diff
@@ -31,31 +30,59 @@ const useStyles = makeStyles(() => ({
...
+        .join(" ");
+    const pendingClasses = _([classes.circle, active ? classes.active : classes.pending])
+        .compact()
+        .join(" ");
+    const completedClasses = _([classes.circle, active ? classes.active : classes.completed])
         .compact()
         .join(" ");
 
-    return <div className={mergeClasses}>{text}</div>;
+    if (isPending) {
```
Comment (eperedo) [inline src/webapp/pages/analysis/AnalysisPage.tsx:72]: Here we're returning the same component but with different className. I think it's easier to read if we have a method to calculate the classes and just return the same component.

```tsx

function StepIcon(...) {
    const buildIconClasses = React.useMemo(() => {
        const baseClasses = [classes.circle];

        if (active) {
            return [...baseClasses, classes.active].join(" ");
        } else if (isPending) {
            return [...baseClasses, classes.pending].join(" ");
        } else if (hasIssues) {
            return [...baseClasses, classes.error].join(" ");
        } else {
            return [...baseClasses, classes.completed].join(" ");
        }
    }, [classes, active, hasIssues, isPending]);

    return <div className={buildIconClasses}>{text}</div>;
}

```

**Review by eperedo — CHANGES_REQUESTED** [review]
Thanks @mariaozamiz. A couple of observations about improving code readability, feel free to ask me any questions about it.

**Review by Ramon-Jimenez — APPROVED** [review]
Thx @mariaozamiz  looks OK to me

**Review by eperedo — APPROVED** [review]

**Pull request merged**